lib/client/api.go

```diff
@@ -398,6 +402,81 @@ func MakeDefaultConfig() *Config {
 	}
 }
 
+// VirtualPathKind is the suffix component for env vars denoting the type of
+// file that will be loaded.
+type VirtualPathKind string
+
+const (
+	// VirtualPathEnvPrefix is the env var name prefix shared by all virtual
+	// path vars.
+	VirtualPathEnvPrefix = "TSH_VIRTUAL_PATH"
+
+	VirtualPathKey        VirtualPathKind = "KEY"
+	VirtualPathCA         VirtualPathKind = "CA"
+	VirtualPathDatabase   VirtualPathKind = "DB"
+	VirtualPathApp        VirtualPathKind = "APP"
+	VirtualPathKubernetes VirtualPathKind = "KUBE"
+)
+
+// VirtualPathParams are an ordered list of additional optional parameters
+// for a virtual path. They can be used to specify a more exact resource name
+// if multiple might be available. Simpler integrations can instead only
+// specify the kind and it will apply wherever a more specific env var isn't
+// found.
+type VirtualPathParams []string
+
+// VirtualPathCAParams returns parameters for selecting CA certificates.
+func VirtualPathCAParams(caType types.CertAuthType) VirtualPathParams {
```

Comment:
If this needs to be kept in sync with the CA types I would put a comment in both locations mentioning that any updates need to be reflected on the other side.

Reply:
Oddly tsh only ever uses the HostCA. I do kind of wonder if this is a problem for new clusters since I think the database CA is meant to be used 

In any case the only thing that would need to be updated is on Machine ID's end if/when we ever needed to pass multiple CAs. Right now we're just using the fallback case (just `TSH_VIRTUAL_PATH_CA`, with no CA type) in https://github.com/gravitational/teleport/pull/12687 since we don't care to disambiguate (at least for now).